pkg/block/mem/adapter.go

```diff
@@ -280,8 +280,15 @@ func (a *Adapter) Walk(_ context.Context, walkOpt block.WalkOpts, walkFn block.W
 	a.mutex.RLock()
 	defer a.mutex.RUnlock()
 
-	fullPrefix := getPrefix(walkOpt)
+	// Walk function should be performed in a lexicographical order
+	keys := make([]string, 0)
 	for k := range a.data {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
```

Comment:
Nice catch! can we add a general kv test for it?

Reply:
This is the block adapter - not the kv driver

Reply:
@itaiad200 
I added some infra to generically test adapters and added the specific scenario. Enabled it for mem and local in the meantime. Please have a look